### PR TITLE
FreeRTOS+TCP Zynq: check frame type before calculating ICMP checksum (v3)

### PR DESCRIPTION
**Pull request by htibosch**
Description
-----------
This PR will replace the earlier #177, of which the branch got lost.

The only significant change is in the function `xNetworkInterfaceOutput()`, in which the frame-type is checked before checking the protocol type.

I the current driver, outgoing ARP frames get a size warning. It doesn't cause incorrect behaviour, except for the size warnings in the logging.

Test Steps
-----------
Define `ipconfigDRIVER_INCLUDED_TX_IP_CHECKSUM` as `1` and send an outgoing ping. The ARP before the PING may not trigger a size warning, and the ICMP checksum shall be calculated correctly.

Related Issue
-----------
None.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AniruddhaKanhere — DISMISSED** [review]

File: FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/portable/NetworkInterface/Zynq/NetworkInterface.c [hunk 1]
```diff
@@ -77,14 +77,14 @@
 #define PHY_REG_01_BMSR			0x01	/* Basic mode status register */
 
 #ifndef iptraceEMAC_TASK_STARTING
-	#define iptraceEMAC_TASK_STARTING()	do { } while( 0 )
+	#define iptraceEMAC_TASK_STARTING()    do {} while( ipFALSE_BOOL )
 #endif
 
 /* Default the size of the stack used by the EMAC deferred handler task to twice
```
Comment (gkwicker) [inline FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/portable/NetworkInterface/Zynq/NetworkInterface.c:83]: This comment no longer matches the constant at line 87.

**Review by gkwicker — CHANGES_REQUESTED** [review]
Fix stack size comment

**Review by AniruddhaKanhere — APPROVED** [review]

**Comment by htibosch** [comment]
This PR has moved to FreeRTOS-Plus-TCP [PR #19](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/19).

**Pull request closed without merging**